js/src/maxImageSize.js

```diff
@@ -2,28 +2,29 @@ export default function maxImageSize(...imageUrls) {
   const dimensions = {};
 
   return new Promise((resolve, reject) => {
-    imageUrls.forEach((url, i) => {
-      const image = new Image();
-
-      image.onerrer = function handleImageError(e) {
-        reject(e);
-      };
+    function handleImageError(e) {
+      reject(e);
+    }
 
-      image.onload = function handleImageLoad() {
-        const { width, height } = this;
+    function handleImageLoad(index) {
+      const { width, height } = this;
 
-        // Use the index in case the URL is somehow duplicated.
-        dimensions[i] = { width, height };
+      // Use the index in case the URL is somehow duplicated.
+      dimensions[index] = { width, height };
 
-        if (Object.keys(dimensions).length >= imageUrls.length) {
-          // We are done, so compute the max width and height and resolve.
-          const values = Object.keys(dimensions).map(key => dimensions[key]);
-          const maxWidth = Math.max(...values.map(value => value.width));
-          const maxHeight = Math.max(...values.map(value => value.height));
-          resolve({ width: maxWidth, height: maxHeight });
-        }
-      };
+      if (Object.keys(dimensions).length >= imageUrls.length) {
+        // We are done, so compute the max width and height and resolve.
+        const values = Object.keys(dimensions).map(key => dimensions[key]);
+        const maxWidth = Math.max(...values.map(value => value.width));
+        const maxHeight = Math.max(...values.map(value => value.height));
+        resolve({ width: maxWidth, height: maxHeight });
+      }
+    }
 
+    imageUrls.forEach((url, i) => {
+      const image = new Image();
+      image.onerror = handleImageError;
+      image.onload = handleImageLoad.bind(image, i);
```

Comment:
You could probably move this functions outside of the Promise too, with some cleverness of making sure that they're still attached to the original promise (passing in `resolve` as a `.bind()` perhaps?).